src/server.ts

```diff
@@ -22,15 +22,24 @@ export interface WebSocketServer extends EventEmitter {
 
 class Server extends EventEmitter {
   private readonly server: http.Server | https.Server
+  private readonly wsServer: WSServer
 
-  constructor (server: http.Server | https.Server) {
+  constructor (server: http.Server | https.Server, opts?: ServerOptions) {
     super()
-
+    opts = opts ?? {}
     this.server = server
+    this.wsServer = new WSServer({
+      server: server,
+      perMessageDeflate: false,
+      verifyClient: opts.verifyClient
+    })
+    this.wsServer.on('connection', this.onWsServerConnection.bind(this))
   }
 
   async listen (addrInfo: { port: number } | number) {
-    return await new Promise<WebSocketServer>(resolve => {
+    return await new Promise<WebSocketServer>((resolve, reject) => {
+      this.server.once('error', (e) => reject(e))
+      this.wsServer.once('error', (e) => reject(e))
       this.once('listening', () => resolve(this))
       this.server.listen(typeof addrInfo === 'number' ? addrInfo : addrInfo.port)
```

Comment:
The WSServer sets up an error listener and also forwards on the listen events from the underlying http(s) server so we can just listen for events on that to resolve the promise.

```suggestion
    return await new Promise<WebSocketServer>((resolve, reject) => {
      this.wsServer.once('error', (e) => reject(e))
      this.wsServer.once('listening', () => resolve(this))
      this.server.listen(typeof addrInfo === 'number' ? addrInfo : addrInfo.port)
```